packages/webamp/js/components/FFTNullsoft.ts

```diff
@@ -0,0 +1,206 @@
+// The Web Audio API's FFT is bad, so this exists now!
+// Taken from https://github.com/WACUP/vis_classic/tree/master/FFTNullsoft
+
+export class FFT {
+  private bitrevtable: number[];
+  private envelope: Float32Array | null;
+  private equalize: Float32Array | null;
+  private temp1: Float32Array;
+  private temp2: Float32Array;
+  private cossintable: Float32Array[];
+
+  // Constants
+  private static readonly TWO_PI = 6.2831853; // 2 * Math.PI
+  private static readonly HALF_PI = 1.5707963268; // Math.PI / 2
+
+  constructor() {
+    // Assuming these are your hardcoded values:
+    const samplesIn = 1024; // hardcoded value
+    const samplesOut = 512; // hardcoded value
+    const bEqualize = true; // hardcoded value
+    const envelopePower = 1.0; // hardcoded value
+    const mode = false; // hardcoded value
+
+    const NFREQ = samplesOut * 2;
+
+    // Initialize the tables and arrays with hardcoded values
+    this.bitrevtable = this.initBitRevTable(NFREQ);
+    this.cossintable = this.initCosSinTable(NFREQ);
+
+    this.envelope =
+      envelopePower > 0
+        ? this.initEnvelopeTable(samplesIn, envelopePower)
+        : null;
```

Comment:
Why do we need this condition? Won't `envelopePower` always be `1.0` and thus always be greater than 0?
```suggestion
    this.envelope = this.initEnvelopeTable(samplesIn, envelopePower);
```